lib/package-managers/npm.js

```diff
@@ -159,78 +205,92 @@ module.exports = {
     /**
      * @param {string} packageName
      * @param {string} currentVersion
-     * @param {boolean} pre
+     * @param {{}} options
      * @returns {Promise}
      */
-    latest(packageName, currentVersion, pre) {
-        return view(packageName, 'dist-tags.latest', currentVersion)
-            .then(version => {
-                // if latest is not a prerelease version, return it
-                // if latest is a prerelease version and --pre is specified, return it
-                if (!isPre(version) || pre) {
-                    return version;
-                // if latest is a prerelease version and --pre is not specified, find the next
-                // version that is not a prerelease
-                } else {
-                    return view(packageName, 'versions', currentVersion)
-                        .then(filterOutPrereleaseVersions)
-                        .then(_.last);
-                }
-            });
+    latest(packageName, currentVersion, options) {
+        return view(packageName, 'versions', currentVersion)
+            .then((versions) => {
+                return filterSatisfiesMinVersion(versions, options.enginesNode);
+            })
+            .then((versions) => {
+                return filterOutPrereleaseVersions(versions, options.pre);
+            })
```

Comment:
Since `filterSatisfiesMinVersion` is not asynchronous, it does not need a separate `then` statement and can be combined with the one for `filterOutPrereleaseVersions`.

(The same can be done for `_.last` for that matter)